modules/core/shared/src/main/scala/gem/syntax/TreeMap.scala

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) 2016-2017 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package gem.syntax
+
+import cats.Foldable
+
+import scala.collection.immutable.TreeMap
+
+final class TreeMapCompanionOps(val self: TreeMap.type) extends AnyVal {
+
+  /** Creates a `TreeMap` from a `List[(A, B)]`, provided an `Ordering[A]`
+    * is available.
+    */
+  def fromList[A: Ordering, B](lst: List[(A, B)]): TreeMap[A, B] =
+    TreeMap(lst: _*)
+
+  /** Creates a `TreeMap` from a `Foldable[(A, B)]`, provided an `Ordering[A]`
+    * is available.
+    */
+  def fromFoldable[F[_], A, B](fab: F[(A, B)])(implicit F: Foldable[F], A: Ordering[A]): TreeMap[A, B] =
+    fromList(F.toList(fab))
+}
+
+trait ToTreeMapCompanionOps {
+  implicit def ToTreeMapCompanionOps(c: TreeMap.type): TreeMapCompanionOps =
+    new TreeMapCompanionOps(c)
+}
+
+object treemapcompanion extends ToTreeMapCompanionOps
```

Comment:
In cats and scalaz this would be called `treemap` and if we had a bedazzlement trait `ToTreeMapOps` we would add it here. Not a big deal but I think it would be ok to fold X and companion X into the same hunk of syntax. Same with `TreeSet` now that I look.

Reply:
Thanks, I am in fact adding a `ToTreeMapOps` and I renamed the object `treemap` there.